src/core/basetypes/MCData.cpp

```diff
@@ -510,44 +534,47 @@ Data * Data::dataWithContentsOfFile(String * filename)
         fclose(f);
         return NULL;
     }
-    
-    buf = (char *) malloc((size_t) stat_buf.st_size);
-    
-    read_items = fread(buf, 1, (size_t)  stat_buf.st_size, f);
-    if ((off_t) read_items != stat_buf.st_size) {
-        free(buf);
-        fclose(f);
+
+    unsigned int length = (unsigned int)stat_buf.st_size;
+    void * bytes = mmap(NULL, length, PROT_READ, MAP_PRIVATE, fileno(f), 0);
+    fclose(f);
+
+    if (bytes == MAP_FAILED) {
         return NULL;
     }
-    
-    data = Data::data();
-    data->takeBytesOwnership(buf, (unsigned int) stat_buf.st_size);
-    
-    fclose(f);
-    
-    return data;
+
+    Data * result = new Data();
```

Comment:
Yes, I can. I do it tomorrow.
Fallback works via calls of reset() and allocate() now. I will rewrite patch to make it more clear.